proposed/logging-generator.md

```diff
@@ -0,0 +1,327 @@
+# Logging Generator for strongly-typed logging messages
+
+## Summary
+
+This is a proposal for a new approach to logging to allow for more easier setup of strongly-typed logging messages. This can be achieved by using a source generator, that gets triggered with a `LoggerMessageAttribute` on partial logging methods, and would be able to either autogenerate the implementation of such method, or produce compile-time diagnostics hinting developers into proper usage of this logging approach.
+
+## Scenarios
+
+Heads up that `LoggingSample1` and `LoggingSample2` show existing approaches to logging, and the remaining will illustrate the samples using the source generator approach with `LoggerMessageAttribute`.
+
+## Background / Existing Approaches
+
+Currently there are two forms possible for doing logging, using `Microsoft.Extensions.Logging`:
+
+```c#
+public class LoggingSample1
+{
+    private ILogger _logger;
+
+    public LoggingSample1(ILogger logger)
+    {
+        _logger = logger;
+    }
+
+    public void LogMethod(string name)
+    {
+        _logger.LogInformation("Hello {name}", name);
+    }
+}
+```
+
+Here are some problems with the `LoggingSample1` sample using `LogInformation`, `LogWarning`, etc.: 
+1. We cannot provide event ID through these APIs
+2. Parameters passed are processed before LogLevel checks
+3. It requires parsing of message string on every use to find templates to substitute
+
+Because of these problems, the more efficient approach recommended today as best practices is to use [LoggerMessage.Define](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/loggermessage?view=aspnetcore-5.0) APIs instead, illustrated below with `LoggingSample2`:
+
+```c#
```

Comment:
nit: this doc uses both `c#` and `csharp` for formatting.  Only the latter gets coloration in the github rendering.

Reply:
> Only the latter gets coloration in the github rendering.

Not true. 
```C#
public class ThisHas
{
    public static bool Color;
}
```

Reply:
Maybe the comment colorization is different than file browser?  Here's what I see:
![image](https://user-images.githubusercontent.com/8918108/113440000-7318fd00-93a0-11eb-995b-8db0bc946324.png)
![image](https://user-images.githubusercontent.com/8918108/113440031-8035ec00-93a0-11eb-8beb-7e61455a8b93.png)